engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -10,302 +10,240 @@ import (
 
 	"github.com/rs/zerolog"
 
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+
 	"github.com/onflow/flow-go/access"
 	"github.com/onflow/flow-go/engine/access/subscription"
-	"github.com/onflow/flow-go/engine/common/rpc"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module/irrecoverable"
-	"github.com/onflow/flow-go/state"
-	"github.com/onflow/flow-go/storage"
-
-	"github.com/onflow/flow/protobuf/go/flow/entities"
 )
 
+// TransactionExpiryForUnknownStatus defines the number of blocks after which
+// a transaction with an unknown status is considered expired.
+// This includes the default transaction expiry time (flow.DefaultTransactionExpiry)
+// plus an additional buffer of 10 blocks to account for potential network delays
+// or reorganization effects.
+const TransactionExpiryForUnknownStatus = flow.DefaultTransactionExpiry + 10
```

Comment:
The buffer really shouldn't be needed. The only ways that the TX would end up unknown are:
1. The AN is behind syncing collections (and indexing), so it doesn't know about the TX yet.
2. The transaction doesn't exist (or is from a past spork)

In the case of 1, the buffer is unlikely to be exactly the time needed to catch up, so is unlikely to prevent an error response.
In the case of 2, we could know exactly `flow.DefaultTransactionExpiry` blocks after the first block searched.

The client can't subscribe before the reference block is produced, since the blockID must be known to generate the transaction ID.

```suggestion
// TransactionExpiryForUnknownStatus defines the number of blocks after which
// a transaction with an unknown status is considered expired.
const TransactionExpiryForUnknownStatus = flow.DefaultTransactionExpiry
```